lib/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -653,6 +657,15 @@ static ParseResult parseReturnOp(OpAsmParser *parser, OperationState *result) {
   return success();
 }
 
+// <operation> ::= `llvm.unreachable` `:` type-list-no-parens
+static ParseResult parseUnreachableOp(OpAsmParser *parser,
+                                      OperationState *result) {
+  Type type;
```

Comment:
This parser shouldn't need to do anything, just inline it into the op definition.